lib/provider.js

```diff
@@ -83,6 +83,53 @@ class StorageProvider {
   select(category, query) {
     throw new GSError(errorCodes.NOT_IMPLEMENTED);
   }
+
+  // Check that given user has access to execute a function
+  //   category - <string>, name of a category that the action is defined upon
+  //   name - <string>, name of an Action to execute
+  //   userId - <common.Uint64>, id of user that executes an Action
+  //   args - <Object>, object that contains arguments of an action
+  //   callback - <function>
+  //     error - <Error>
+  //     permission - <boolean>
+  // eslint-disable-next-line no-unused-vars
+  checkPermission(category, name, userId, args, callback) {
+    throw new GSError(errorCodes.NOT_IMPLEMENTED);
+  }
+
+  // Execute Action
+  //   category - <string>, name of a category that the action is defined upon
+  //   name - <string>, name of an Action to execute
+  //   userId - <common.Uint64>, id of user that executes an Action
+  //   args - <Object>, object that contains arguments of an action
+  //   callback - <function>
+  //     error - <Error>
+  //     message - <string>
+  execute(category, name, userId, args, callback) {
+    const form = `${category}.${name}`;
+    const action = this.gs.schema.action.get(form);
+    if (!action) {
+      callback(new GSError(errorCodes.NO_SUCH_ACTION));
```

Comment:
You haven't added an error message as requested in https://github.com/metarhia/globalstorage/pull/259#discussion_r230128561, even though you have resolved the discussion there.